Model/Media.php

```
@@ -643,8 +642,10 @@ public function getCategory()
     /**
      * @param CategoryInterface $category|null
      */
-    public function setCategory(CategoryInterface $category = null)
+    public function setCategory($category = null)
     {
-        $this->category = $category;
+        if($category instanceof CategoryInterface) {
```

This is way too silent. You should throw an exception instead.